### ISSUE TITLE
devcontainer build with --push true failed when use image property

### ISSUE THREAD

**Issue by stickmy** [devcontainers/cli#370]
#### devcontainer.json with image property

```json
{
    "name": "kaitian-devcontainer",
    "image": "mcr.microsoft.com/devcontainers/javascript-node:14",
    "features": {
        "ghcr.io/stickmy/merchant-legacy-dev-features/kaitian:1.0.1": {}
    }
}
```

#### command

```bash
devcontainer build --workspace-folder . --image-name ghcr.io/stickmy/kaitian-devcontainer:1.0.0 --push true
```
The error message:
```
=> ERROR pushing vsc-kaitian-devcontainer-27da8ffe3eebb26605196a518a9f1f

ERROR: denied: requested access to the resource is denied
{"outcome":"error","message":"Command failed: docker buildx build --push --build-context dev_containers_feature_content_source=/var/folders/3n/0hqblg7d3mq2q0nlzkj31pgr0000gp/T/devcontainercli/container-features/0.27.1-1673115040210 --build-arg _DEV_CONTAINERS_BASE_IMAGE=mcr.microsoft.com/devcontainers/javascript-node:14 --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp --target dev_containers_target_stage -t vsc-kaitian-devcontainer-27da8ffe3eebb26605196a518a9f1f22-features -f /var/folders/3n/0hqblg7d3mq2q0nlzkj31pgr0000gp/T/devcontainercli/container-features/0.27.1-1673115040210/Dockerfile.extended /var/folders/3n/0hqblg7d3mq2q0nlzkj31pgr0000gp/T/devcontainercli/empty-folder","description":"An error occurred building the container."}
```

But the build works with `build` property

```json
{
    "name": "kaitian-devcontainer",
    "build": {
        "dockerfile": "Dockerfile",
        "args": {
            "VARIANT": "14"
        }
    },
    "features": {
        "ghcr.io/stickmy/merchant-legacy-dev-features/kaitian:1.0.1": {}
    }
}
```

**Dockerfile**

```
ARG VARIANT="14"
FROM mcr.microsoft.com/devcontainers/javascript-node:${VARIANT}
```

**Comment by jimkoen** [comment]
Same issue here, any update?

**Comment by chrmarti** [comment]
Does `docker push ghcr.io/stickmy/kaitian-devcontainer:1.0.0` work? Also see https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry on authentication to ghcr.io.

**Comment by stickmy** [comment]
> Does `docker push ghcr.io/stickmy/kaitian-devcontainer:1.0.0` work? Also see https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry on authentication to ghcr.io.

Yes, it works. Already logged in the `ghcr.io` container registry. I personally don't think it's caused by this.

Is the error message: `=> ERROR pushing vsc-kaitian-devcontainer-27da8ffe3eebb26605196a518a9f1f` means the image being pushed is `vsc-kaitian-devcontainer-27da8ffe3eebb26605196a518a9f1f`?

**Comment by froland** [comment]
Hello,
I got the same issue: building without pushing works but pushing doesn't.
When I inspect the built image, I get 2 "RepoTags" values: the one I want and another one which seems generated "vsc-{something}-features:latest".
When pushing, it seems like it tries to push to "docker.io/library/vsc-{something}-features", for which I, obviously, don't have permissions as this path doesn't include my hub.docker.io username.

**Comment by chrmarti** [comment]
It seems to push to all tags specified (reading https://github.com/docker/buildx/issues/396). We could omit the generated tag when running `build` with a user-specified tag.

**Comment by froland** [comment]
Hello,
In my devcontainer.json, I've replaced the "image" attribute by a "build" one and moved the image reference to a Dockerfile.
Now the `devcontainer build ...` command behaves correctly. It seems to be a corner case where the tags taken into account.

**Comment by msclock** [comment]
> Hello, In my devcontainer.json, I've replaced the "image" attribute by a "build" one and moved the image reference to a Dockerfile. Now the `devcontainer build ...` command behaves correctly. It seems to be a corner case where the tags taken into account.

This solves my situation. haha.
